chrome/content/zotero/xpcom/syncedSettings.js

```diff
@@ -175,11 +209,13 @@ Zotero.SyncedSettings = (function () {
 			version = parseInt(version);
 			
 			if (hasCurrentValue) {
-				var sql = "UPDATE syncedSettings SET value=?, version=?, synced=? "
-					+ "WHERE setting=? AND libraryID=?";
-				yield Zotero.DB.queryAsync(
-					sql, [JSON.stringify(value), version, synced, setting, libraryID]
-				);
+				var sql = "UPDATE syncedSettings SET " + (version > 0 ? "version=?, " : "") + 
+					"value=?, synced=? WHERE setting=? AND libraryID=?";
+				var args = [JSON.stringify(value), synced, setting, libraryID];
+				if (version > 0) {
+					args.splice(0, 0, version)
```

Comment:
Yeah, I usually continue the SQL and params after adding any conditional ones. But `unshift` would be fine too.